gubernator/github/cron.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Periodically synchronize our Datastore view of PRs with Github.
+
+Various things can cause the local status of a PR to diverge from upstream:
+dropped hooks from bugs in the app, upstream GitHub bugs (webhooks aren't
+guaranteed!), or a repo *just* starting sending hooks to Gubernator.
+
+Divergent PR state make the PR dashboard less useful, since old PRs accumulate
+and clutter out real items, decreasing signal-to-noise ratio and user trust.
+
+To handle these, on a regular schedule we perform a reconciliation step:
+- for each repository that we're tracking:
+  - A = all open PRs from Datastore
+  - B = all open PRs from Github
+  - A-B is the set of improperly open PRs. For each PR, add a synthetic
+    webhook event to Datastore with state=closed, and reprocess.
+  - B-A is the set of improperly closed or missing PRs. Again, inject a
+    synthetic webhook with the details received from GitHub and reprocess.
+
+This requires a Github token set like other secrets with /config in the root.
+Total token usage is low: number of open PRs / 30 PRs per list call.
```

Comment:
You should be able to get as many as 100 PRs per list call . https://developer.github.com/v3/guides/traversing-with-pagination/#changing-the-number-of-items-received

Reply:
That's for the search API.

Reply:
It seems to work when using listing pull requests. It isn't well documented, but I think 100 items per page is the limit for most if not all of the v3 API.